File: backend/Models/Instance.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace prid_2425_f06.Models
{
    public class Instance
    {   
        private readonly FormContext? _context;
        public Instance(FormContext context){
            this._context = context;
        }
        public Instance(){}
        [Key]
        public int InstanceId { get; set; }
        [ForeignKey(nameof(Form))]
        public int FormId { get; set; } 
        public Form Form { get; set; } = null!;
        public int UserId { get; set; } 
     
        public DateTimeOffset Started { get; set; } = DateTimeOffset.Now;
        public DateTimeOffset? Completed { get; set; }

        public ICollection<Answer> Answers { get; set; } = new HashSet<Answer>();
        
        
    }


}

File: frontend/src/app/models/instance.ts
import { extend } from 'lodash-es';
import { Form } from './form';

export class Instance {
    instanceId: number = 0;
    formId: number = 0;
    userId: number = 0;
    started?: Date;
    completed?: Date;
    

}



export class InstanceWithFormDetailed extends Instance {
    form? : Form;

}